packages/lesmis/src/utils/useUrlSearchParams.js

```diff
@@ -8,18 +8,39 @@ import { useHistory, useLocation } from 'react-router-dom';
 export const useUrlSearchParams = () => {
   const history = useHistory();
   const location = useLocation();
-  const params = new URLSearchParams(location.search);
+  const urlParams = new URLSearchParams(location.search);
 
   const setParams = newParams => {
     Object.entries(newParams).forEach(([key, param]) => {
       if (param === null || param === undefined) {
-        params.delete(key);
+        urlParams.delete(key);
       } else {
-        params.set(key, param.toString());
+        urlParams.set(key, param.toString());
       }
     });
-    history.push({ ...history.location, search: params.toString() });
+
+    if (location.search !== urlParams.toString()) {
+      history.push({ ...location, search: urlParams.toString() });
+    }
   };
 
+  const params = {};
+
+  urlParams.forEach((value, key) => {
```

Comment:
What kind of data structure is `urlParams`? Is the iterator just a weird custom thing?

Reply:
I'm not sure I understand the question. [urlSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) is part of the core javascript api it returns an iterable object. An iterator is a javascript thing - it's not custom in the sense that I made it.

Reply:
I guess my question is was why the parameters are `value, key`, which is quite unusual. Thanks for the link to the doc - it seems like the "natural" iterator over the data structure is the more expected `([key, value])` (like Object.entries), but `.forEach` is `(value, key)`. Strange that it would have different formats of iterable depending on which `for` you use, but I guess that's javascript!